File: src/_03_BehavioralPattern/_03_05_Mediator/java/after/Guest.java
package _03_BehavioralPattern._03_05_Mediator.java.after;

public class Guest {

  private Integer id;

  private FrontDesk frontDesk = new FrontDesk();

  public Guest(int id) {
    this.id = id;
  }

  public int getId() {
    return this.id;
  }

  public void setId(int id) {
    this.id = id;
  }
  
  public void getTowers(int numberOfTowers) {
    this.frontDesk.getTowers(this, numberOfTowers);
  }

  public void dinner(String menu) {
    this.frontDesk.dinner(this, menu);
  }
  
}

File: src/_03_BehavioralPattern/_03_05_Mediator/Guest.js
/**
 * @typedef { import("./FrontDesk").default } FrontDesk
 */

export default class Guest {
  /** @type { FrontDesk } */
  _frontDesk;

  /** @type { number } */
  _id;

  /** 
   * @param { FrontDesk } frontDesk 
   * @param { number } id
   */
  constructor(frontDesk, id) {
    this._frontDesk = frontDesk;
    this._id = id;
  }

  /** @returns { number } */
  getId() {
    return this._id;
  }

  /** @param { number } numberOfTowers */
  getTowers(numberOfTowers) {
    this._frontDesk.getTowers(this, numberOfTowers);
  }

  /** @param { string } menu */
  dinner(menu) {
    this._frontDesk.dinner(this, menu);
  }
}